core/src/main/java/io/questdb/cairo/PartitionBy.java

```diff
@@ -183,36 +180,101 @@ public static boolean isPartitioned(int partitionBy) {
     }
 
     public static long parsePartitionDirName(@NotNull CharSequence partitionName, int partitionBy) {
-        return parsePartitionDirName(partitionName, 0, partitionName.length(), partitionBy);
-    }
-
-    public static long parsePartitionDirName(@NotNull CharSequence partitionName, int lo, int hi, int partitionBy) {
+        CharSequence fmtStr;
+        int limit = -1;
         try {
-            return getPartitionDirFormatMethod(partitionBy).parse(partitionName, lo, hi, null);
-        } catch (NumericException e) {
-            final CairoException ee = CairoException.critical(0).put('\'');
+            DateFormat fmtMethod;
             switch (partitionBy) {
                 case DAY:
-                    ee.put(DAY_FMT);
-                    break;
-                case WEEK:
-                    ee.put(WEEK_FMT);
+                    fmtMethod = DAY_FORMAT;
+                    fmtStr = DAY_PATTERN;
                     break;
                 case MONTH:
-                    ee.put(MONTH_FMT);
+                    fmtMethod = MONTH_FORMAT;
+                    fmtStr = MONTH_PATTERN;
                     break;
                 case YEAR:
-                    ee.put(YEAR_FMT);
+                    fmtMethod = YEAR_FORMAT;
+                    fmtStr = YEAR_PATTERN;
                     break;
                 case HOUR:
-                    ee.put(HOUR_FMT);
+                    fmtMethod = HOUR_FORMAT;
+                    fmtStr = HOUR_PATTERN;
+                    break;
+                case WEEK:
+                    fmtMethod = WEEK_FORMAT;
+                    fmtStr = WEEK_PATTERN;
+                    break;
+                case NONE:
+                    fmtMethod = DEFAULT_FORMAT;
+                    fmtStr = partitionName;
                     break;
+                default:
+                    throw new UnsupportedOperationException("partition by " + partitionBy + " does not have date format");
+            }
+            limit = fmtStr.length();
+            if (partitionName.length() < limit) {
+                throw expectedPartitionDirNameFormatCairoException(partitionName, partitionName.length(), partitionBy);
             }
-            ee.put("' expected");
-            throw ee;
+            return fmtMethod.parse(partitionName, 0, limit, null);
+        } catch (NumericException e) {
+            if (partitionBy == PartitionBy.WEEK) {
+                // maybe the user used a timestamp, or a date, string.
+                int local_limit = DAY_PATTERN.length();
```

Comment:
nit: Java code should use camel case:
```suggestion
                int localLimit = DAY_PATTERN.length();
```